spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/Hack.java

```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.reactor;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.reactivestreams.Processor;
+import org.reactivestreams.Publisher;
+import reactor.core.CoreSubscriber;
+import reactor.core.Disposable;
+import reactor.core.Fuseable;
+import reactor.core.Scannable;
+import reactor.core.publisher.ConnectableFlux;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxOperator;
+import reactor.core.publisher.GroupedFlux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoOperator;
+import reactor.core.publisher.Operators;
+import reactor.core.publisher.ParallelFlux;
+import reactor.util.annotation.Nullable;
+
+/**
+ * Helper class for reactor ON_EACH instrumentation. Reduces number of sleuth operators
+ * time in the reactive chain during Assembly by using {@code Scannable.Attr.RUN_STYLE}.
+ * Below is the example of what it does <pre>{@code
+ *	Mono.just(0) // (-)
+ * 	.map(it -> 1) // (-)
+ * 	.flatMap(it ->
+ * 		Mono.just(it) // (-)
+ * 	) // (-)
+ * 	.flatMapMany(it ->
+ * 		asyncPublisher(it) // (+)(*)
```

Comment:
This is an interesting one to explain as to the naked eye, it seems like it might drop the initial subscribe context. The other steps mentioned are more obvious and probably don't need more notes. You could add comments here or maybe a `<h3>Why asyncPublisher re-captures context from Brave</h3>` to free you from questions later.

Reply:
probably, to be more accurate this 
>(*) - captures brave context at subscription and propagates it  

could be rephrased to  
>(*) - captures brave context if it differs from what was captured before at subscription and propagates it. As far as check is performed on `Subscriber` `Context` it allocates `Publisher` to access it